### PR TITLE
SPL log,ln,floor and pow functionality

### PR DESCRIPTION
**Pull request by venkateshamatam**
# Description
SPL log,ln,floor and pow functionality

# Testing
- Tested with queries like 
`city=Boston | stats count AS Count BY http_status | eval squared_http_status=pow(http_status, 2)`
`city=Boston | stats count AS Count BY http_status | eval log_2_http_status=log(http_status, 2)`
`city=Boston | stats count AS Count BY http_status | eval log_2_http_status=log(http_status, 10)`
`city=Boston | stats count AS Count BY http_status | eval log_2_http_status=ln(http_status)`
`city=Boston | stats count AS Count BY http_status | eval log_2_http_status=floor(http_status)`
`city=Boston | stats count AS Count BY http_status | eval log_2_http_status=floor(-1.9)`

Will add test cases to CICD
# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

**Comment by Jiale-Fang** [comment]
Can you also update this comments?
https://github.com/siglens/siglens/blob/55294d1f8b06b24cb9ff85c22831e0c7ae96cf7c/pkg/segment/aggregations/segaggs.go#L125-L135

File: pkg/segment/structs/evaluationstructs.go [hunk 1]
```diff
@@ -1239,6 +1239,23 @@ func (self *NumericExpr) Evaluate(fieldToValue map[string]utils.CValueEnclosure)
 			return math.Abs(left), nil
 		case "ceil":
 			return math.Ceil(left), nil
+		case "log":
+			if left < 0 {
+				return -1, fmt.Errorf("NumericExpr.Evaluate: Negative values cannot be used for logarithm: %v", left)
+			}
+			if right != 2 && right != 10 {
+				return -1, fmt.Errorf("NumericExpr.Evaluate: Logarithm base must be either 2 or 10: %v", right)
+			}
```
Comment (AndrewHess) [inline pkg/segment/structs/evaluationstructs.go:1248]: Is this in the splunk documentation? I'm not finding it

Comment (venkateshamatam) [reply pkg/segment/structs/evaluationstructs.go:1248]: yes, it's here https://docs.splunk.com/Documentation/SCS/current/SearchReference/MathematicalFunctions#log.28.26lt.3Bnum.26gt.3B.2C.26lt.3Bbase.26gt.3B.29

Comment (AndrewHess) [reply pkg/segment/structs/evaluationstructs.go:1248]: Those are just examples where `2` and `10` are valid values, but I don't see anything indicating all other values are invalid

**Pull request closed without merging**